app/qml/editor/inputqrcodereader.qml

```diff
@@ -72,5 +74,29 @@ AbstractEditor {
     }
   }
 
-  onRightActionClicked: root.importDataRequested()
+  onRightActionClicked: {
+    if ( root.parent.readOnly ) return
+
+    let page = root.parent.formView.push( readerComponent, {} )
+    page.forceActiveFocus()
+  }
+
+  Component {
+    id: readerComponent
+
+    CodeScanner {
+      id: codeScannerPage
+
+      focus: true
+
+      onBackButtonClicked: {
+        root.parent.formView.pop()
```

Comment:
Can you please declare the stackview as a property? 
`property StackView formView`